test/unit/dataTierApplicationWebviewController.test.ts

```diff
@@ -0,0 +1,2534 @@
+            expect(result.autoConnected).to.be.false; // Already was connected
+        });
+
+        test("handles exception during initialization gracefully", async () => {
+            // Setup
+            connStoreStub.getRecentlyUsedConnections.throws(new Error("Store error"));
+
+            createController();
+
+            // Execute
+            const handler = requestHandlers.get(InitializeConnectionWebviewRequest.type.method);
+            expect(handler, "InitializeConnection handler should be registered").to.exist;
+
+            const result = await handler!({
+                initialProfileId: "profile1",
+            });
+
+            // Verify - when store fails, listConnections catches it and returns empty array
+            // initializeConnection then returns empty array with no match found
+            expect(result.connections).to.exist;
+            expect(result.connections).to.be.an("array").that.is.empty;
+            expect(result.autoConnected).to.be.false;
+            expect(result.selectedConnection).to.be.undefined;
+            // No error message is set because listConnections handles the error internally
+        });
+
+        test("prioritizes profile ID match over server name match", async () => {
+            // Setup
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            const recentConns: any[] = [
+                {
+                    id: "profile1",
+                    server: "server1",
+                    database: "db1",
+                    authenticationType: 2, // SqlLogin
+                },
+                {
+                    id: "profile2",
+                    server: "server1",
+                    database: "db2",
+                    authenticationType: 1, // Integrated
+                },
+            ];
+
+            connStoreStub.getRecentlyUsedConnections.returns(recentConns);
+            sandbox.stub(connectionManagerStub, "activeConnections").get(() => ({}));
+
+            createController();
+
+            // Execute - both profile ID and server name match, profile ID should win
+            const handler = requestHandlers.get(InitializeConnectionWebviewRequest.type.method);
+            const result = await handler!({
+                initialProfileId: "profile2",
+                initialServerName: "server1",
+            });
+
+            // Verify - should match profile2 by ID, not profile1 by server
+            expect(result.selectedConnection).to.exist;
+            expect(result.selectedConnection?.profileId).to.equal("profile2");
+            expect(result.selectedConnection?.database).to.equal("db2");
+        });
+    });
+
+    suite("Telemetry", () => {
+        let startActivityStub: sinon.SinonStub;
+        let endStub: sinon.SinonStub;
+        let endFailedStub: sinon.SinonStub;
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        let mockActivity: any;
```

Comment:
avoid `any`; this looks like it could've been cast as an `ActivityObject` instead